### PR TITLE
NuGet automatic package restore

### PR DESCRIPTION
**Pull request by sharwell**
This is the first part of some work to clean up the build process.

* Migrate to NuGet automatic package restore (depends on related PRs)
* Copy plugins to the output directories using properly tracked items, so Build/Rebuild/Clean always work exactly as expected

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -9,5 +9,10 @@ indent_style = space
 indent_size = 4
 trim_trailing_whitespace = true
 
+[*.{csproj,vcxproj}]
+indent_style = space
+indent_size = 2
+insert_final_newline = false
```
Comment (sharwell) [inline .editorconfig]: :memo: This is the format Visual Studio uses when it makes changes to these files. Now hand-editing is consistent too.

File: GitExtensions/copy_plugins.cmd [hunk 2]
```diff
@@ -1,11 +0,0 @@
-@echo off
-set config=%1
-md Plugins\
-echo Microsoft.TeamFoundation.WorkItemTracking.Client.DataStoreLoader.dll > exclude.txt
-echo Microsoft.WITDataStore.dll >> exclude.txt
```
Comment (sharwell) [inline GitExtensions/copy_plugins.cmd]: ❗️ Need to modify the copy to account for these exclusions.

Comment (sharwell) [inline GitExtensions/copy_plugins.cmd]: ➡️ Now fixed

**Comment by jbialobr** [comment]
Thank you @sharwell. I merged your changes in #4078.
While it all worked in VS, there were problems when building with msbuild with nuget below 4.3 which is currently not installed on AppVeyor CI. I added manual restoring nuget packages to our Build scripts (we will be able to remove it when AppVeyor installs updates).

**Comment by sharwell** [comment]
@jbialobr You can use the Visual Studio 2017 image on AppVeyor if you want to have the new behavior now.

**Comment by jbialobr** [comment]
I previously tried with `os: Visual Studio 2017`, but with no luck:
https://ci.appveyor.com/project/jbialobr/gitextensions/build/2.50.02.370#L27
https://github.com/jbialobr/gitextensions/commit/4220fc3322bf129c2545949a5b83abcc57fdfc41#diff-180360612c6b8c4ed830919bbb4dd459

With `image: Visual Studio 2017` there is the same otuput:
https://ci.appveyor.com/project/jbialobr/gitextensions/build/2.50.02.384#L30
https://github.com/gitextensions/gitextensions/commit/b37191937f895ce30cfc8b9e68db52cdc0bb8ab1

**Pull request closed without merging**